pkg/nsx/services/vpc/vpc.go

```diff
@@ -405,6 +405,22 @@ func (s *VPCService) GetNSXLBSNATIP(vpc model.Vpc) (string, error) {
 	return tier1UpLinkIP, nil
 }
 
+func (s *VPCService) GetNSXLBSNATIPForDTGW(vpc model.Vpc) (string, error) {
+	log.V(2).Info("Fetching NSX LB SNAT IP from DTGW", "VPC", *vpc.Id)
+	_, err := common.ParseVPCResourcePath(*vpc.Path)
+	if err != nil {
+		return "", err
+	}
+	realizeService := realizestate.InitializeRealizeState(s.Service)
+	lbIP, err := realizeService.GetPolicyServiceInterfaceSNATIP(*vpc.Path + "/service-interface")
```

Comment:
I do not think we should use this API to get the LB SNAT IP

Reply:
fixed